src/java/htsjdk/samtools/BAMIndexer.java

```diff
@@ -55,6 +55,10 @@
      */
     public BAMIndexer(final File output, final SAMFileHeader fileHeader) {
 
+        if (fileHeader.getSortOrder() != SAMFileHeader.SortOrder.coordinate) {
+            throw new SAMException("BAM file be coordinate-sorted for indexing.");
```

Comment:
Lost a word here,  BAM file **must** be.

Could you also include what the sort order is in the error message?